### PR TITLE
Add system packages and an app for Go

### PR DESCRIPTION
**Pull request by alexellis**
## Description

Add system packages and an app for Go

```bash
arkade system install go --version 1.18
```

## Motivation and Context

#654 

## How Has This Been Tested?

```bash
$ go build && ./arkade system install go --path /tmp/go-118
Installing Go to /tmp/go-118
Installing version: go1.18 for: amd64
Downloading from: https://go.dev/dl/go1.18.linux-amd64.tar.gz
135.14 MiB / 135.14 MiB [---------------------------------------------------------------------------] 100.00%
Downloaded to: /tmp/go1.18.linux-amd64.tar.gz
Unpacking Go to: /tmp/go-118/go
2022/04/05 09:29:41 extracted tarball into /tmp/go-118: 11753 files, 1378 dirs (2.254868602s)

export PATH=$PATH:/tmp/go-118/go/bin:$HOME/go/bin
export GOPATH=$HOME/go/
$ /tmp/go-118/go/bin/go version
go version go1.18 linux/amd64
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Documentation

I've updated the README

## Checklist:

- [x] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

File: cmd/system/system.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	"github.com/spf13/cobra"
+)
+
+func MakeSystem() *cobra.Command {
+
+	command := &cobra.Command{
+		Use:     "system",
+		Short:   "System apps",
+		Long:    `Apps for systems.`,
+		Aliases: []string{"v"},
```
Comment (Shikachuu) [inline cmd/system/system.go:17]: Why `v`?

File: pkg/archive/untar_nested.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	"path/filepath"
+	"time"
+)
+
+// Untar reads the gzip-compressed tar file from r and writes it into dir.
+func UntarNested(r io.Reader, dir string) error {
+	return untarNested(r, dir)
+}
+
+func untarNested(r io.Reader, dir string) (err error) {
```
Comment (Shikachuu) [inline pkg/archive/untar_nested.go:19]: Should we move this functions under `pkg/archive/untar.go`?
I think it makes much more sense.

File: cmd/system/go.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+	command.PreRunE = func(cmd *cobra.Command, args []string) error {
+
+		return nil
+	}
+
+	command.RunE = func(cmd *cobra.Command, args []string) error {
+		installPath, _ := cmd.Flags().GetString("path")
+		version, _ := cmd.Flags().GetString("version")
+		fmt.Printf("Installing Go to %s\n", installPath)
```
Comment (Shikachuu) [inline cmd/system/go.go:43]: Should we introduce `--quiet` flag as we did earlier for binary downloads?

File: cmd/system/system.go (hunk 1 above)
Comment (alexellis) [reply cmd/system/system.go:17]: I copied from the venafi app so we can change that to `s`

File: pkg/archive/untar_nested.go (hunk 2 above)
Comment (alexellis) [reply pkg/archive/untar_nested.go:19]: No I don't see any reason to have one file for this.

File: cmd/system/go.go (hunk 3 above)
Comment (alexellis) [reply cmd/system/go.go:43]: I'm going to keep things verbose for the preview stages, to make it simpler / less work for everyone.

Then we can come back and fine tune.

**Comment by alexellis** [comment]
@Shikachuu do you want to have a go at the containerd recipe and prometheus next?

@richardcase volunteered to take the firecracker CLI

I didn't use a Go template in this Go app, but if we need it we can bring that back in for the additional apps we're working with.

**Comment by Shikachuu** [comment]
@alexellis sure I'll play around with containerd!

**Pull request merged**